internal/fnruntime/container.go

```diff
@@ -175,11 +176,20 @@ func NewContainerEnvFromStringSlice(envStr []string) *runtimeutil.ContainerEnv {
 }
 
 // AddDefaultImagePathPrefix adds default gcr.io/kpt-fn/ path prefix to image if only image name is specified
-func AddDefaultImagePathPrefix(image string) string {
+func AddDefaultImagePathPrefix(ctx context.Context, image string) string {
+	segments := strings.Split(image, ":")
+	if len(segments) == 4 {
+		// Porch function
+		functionName := strings.Join(segments[1:], ":")
+		function, err := porch.FunctionGetter{Ctx: ctx}.Get(functionName, segments[0])
```

Comment:
I think golang style frowns upon putting context into structs but I also acknowledge we do it in kpt somewhat routinely because cobra seems to force that pattern.

consider whether in this case it is avoidable.

Reply:
Good to know! moved the context to the function parameter.